src/Lean/Meta/InferType.lean

```diff
@@ -279,7 +281,7 @@ mutual
   `isProofQuickApp f n` is an "approximate" predicate which returns `LBool.true`
    if `f` applied to `n` arguments is a proof. -/
 private partial def isProofQuickApp : Expr → Nat → MetaM LBool
-  | .const c lvls,   arity   => do let constType ← inferConstType c lvls; isArrowProposition constType arity
+  | .const c _,      arity   => do let cinfo ← getConstInfo c; isArrowProposition cinfo.type arity
```

Comment:
This change is making the function less precise. I am also bothered that the universes are ignored here but taken into account in other functions above. 
I am also concerned that the change may improve performance in some examples (e.g., core lean), but affect the performance in others.